### PR TITLE
Add PR to county and state view.

### PR DESCRIPTION
**Pull request by sarah-colq**
Add Puerto Rico to state and county choropleth maps by using shapefiles.  Remove centroids and change order of transformations to properly reflect what happens in the Python version.  Use megacounties properly for counties by making them the entire state as a transparent background.  Update ubuntu version to 20.04 because it has a more updated version of gdal and projection.

### CODE REVIEW

**Comment by sgsmob** [comment]
I am going to hold off on reviewing until there is a PR description because I am having a hard time understanding what I should be looking for in this PR.

**Comment by sarah-colq** [comment]
@sgsmob Sorry, we had some projection issues that @chinandrew and I are fixing (and I can only test here because it's an Ubuntu specific issue and I'm on a Mac).  I'll clarify when it's done.

**Comment by sarah-colq** [comment]
@sgsmob @capnrefsmmat this is ready for review now.

**Review by krivard — APPROVED** [review]
CI change lgtm.

File: .github/workflows/r_ci.yml
Comment (sgsmob) [inline .github/workflows/r_ci.yml:19]: Can you explain in the PR description what the purpose of changing these are?

File: R-packages/covidcast/R/plot.R
Comment (sgsmob) [inline R-packages/covidcast/R/plot.R:295]: Seems better to delete this line and use `ggplot2::aes` below

File: R-packages/covidcast/R/plot.R
Comment (sgsmob) [inline R-packages/covidcast/R/plot.R:325]: This formatting is weird since the call to `filter()` should fit on one line.

File: R-packages/covidcast/R/plot.R
Comment (sgsmob) [inline R-packages/covidcast/R/plot.R:327]: These arguments should be indented

File: R-packages/covidcast/R/plot.R
Comment (sgsmob) [inline R-packages/covidcast/R/plot.R:366]: What is this `substr()` call doing?

File: R-packages/covidcast/R/plot.R
Comment (sgsmob) [inline R-packages/covidcast/R/plot.R:381]: same question

File: R-packages/covidcast/R/plot.R
Comment (sgsmob) [inline R-packages/covidcast/R/plot.R:406]: Can get rid of this too

File: R-packages/covidcast/R/plot.R
Comment (sgsmob) [inline R-packages/covidcast/R/plot.R:701]: Are these derived from some external source or did you just tinker around to find these values?  Would be good to give more context

File: R-packages/covidcast/R/plot.R
Comment (sgsmob) [inline R-packages/covidcast/R/plot.R:710]: Please add a brief comment (no need to make it fully documented) about the purpose of these `shift_X` functions

File: R-packages/covidcast/R/plot.R
Comment (sgsmob) [inline R-packages/covidcast/R/plot.R:717]: is it actually necessary to add and subtract 0 here?

File: R-packages/covidcast/R/plot.R
Comment (sgsmob) [inline R-packages/covidcast/R/plot.R:748]: add space after `if`

File: R-packages/covidcast/R/plot.R
Comment (sgsmob) [inline R-packages/covidcast/R/plot.R:749]: no need to break this call to `filter` up over multiple lines

File: R-packages/covidcast/R/plot.R
Comment (sgsmob) [inline R-packages/covidcast/R/plot.R:756]: Add a newline at the end of the file

**Review by sgsmob — COMMENTED** [review]
Mostly looks good but I have a lot of questions since I don't understand this code base.

File: R-packages/covidcast/R/plot.R
Comment (sarah-colq) [reply R-packages/covidcast/R/plot.R:366]: It's getting the last two characters; the names are shown as 'Allegheny, PA' and the state is needed.

File: R-packages/covidcast/R/plot.R
Comment (sarah-colq) [reply R-packages/covidcast/R/plot.R:381]: First two letters have the state for hrr.

File: R-packages/covidcast/R/plot.R
Comment (sarah-colq) [reply R-packages/covidcast/R/plot.R:717]: Nope.  Removed!

File: R-packages/covidcast/R/plot.R
Comment (sarah-colq) [reply R-packages/covidcast/R/plot.R:710]: Added!

File: R-packages/covidcast/R/plot.R
Comment (sarah-colq) [reply R-packages/covidcast/R/plot.R:701]: Context given.

File: R-packages/covidcast/R/plot.R
Comment (sarah-colq) [reply R-packages/covidcast/R/plot.R:406]: It's referenced a lot and the code is less messy this way, I'd rather leave it.

File: R-packages/covidcast/R/plot.R
Comment (sarah-colq) [reply R-packages/covidcast/R/plot.R:295]: It's referenced a lot and the code is less messy this way, I'd rather leave it. (same as other comment)

File: R-packages/covidcast/R/plot.R
Comment (sarah-colq) [reply R-packages/covidcast/R/plot.R:325]: Better?

File: R-packages/covidcast/R/plot.R
Comment (sgsmob) [reply R-packages/covidcast/R/plot.R:365]: Can you add this as a comment please?

File: R-packages/covidcast/R/plot.R
Comment (sgsmob) [reply R-packages/covidcast/R/plot.R:380]: Can you add this as a comment here too?

**Review by sgsmob — APPROVED** [review]
Approving pending a few comment adds

**Comment by sarah-colq** [comment]
@sgsmob comments added!

**Comment by sarah-colq** [comment]
@capnrefsmmat Just waiting for you, no rush :)

File: R-packages/covidcast/R/plot.R
Comment (capnrefsmmat) [inline R-packages/covidcast/R/plot.R:270]:
```suggestion
  if (attributes(x)$metadata$geo_type == "county") {
```

File: R-packages/covidcast/R/plot.R
Comment (capnrefsmmat) [inline R-packages/covidcast/R/plot.R:274]:
```suggestion
      color = ifelse(paste0(STATEFP,"000") %in% geo,
                     col_fun(val[paste0(STATEFP, "000")], alpha = alpha),
                     missing_col))
```

File: R-packages/covidcast/R/plot.R
Comment (capnrefsmmat) [inline R-packages/covidcast/R/plot.R:717]:
```suggestion
shift_pr = function(map_df) {
```

File: R-packages/covidcast/R/plot.R
Comment (capnrefsmmat) [inline R-packages/covidcast/R/plot.R:755]:
```suggestion
  if ("is_state" %in% colnames(main_df)) {
```

**Review by capnrefsmmat — APPROVED** [review]
Looks fine to me, provided all tests pass and `devtools::check()` doesn't report any new complaints (there should be some existing ones about the dataset documentation and our use of `states.abb`). Just a couple nitpicking formatting suggestions

**Pull request merged**
